redis/credentials.py

```diff
@@ -0,0 +1,50 @@
+from typing import Callable, Optional
+
+
+class CredentialsProvider:
+    def __init__(
+        self,
+        username: str = "",
+        password: str = "",
+        supplier: Optional[Callable] = None,
+        *args,
+        **kwargs,
+    ):
+        """
+        Initialize a new Credentials Provider.
+        :param supplier: a supplier function that returns the username and password.
+                         def supplier(arg1, arg2, ...) -> (username, password)
+                         For examples see examples/connection_examples.ipynb
+        :param args: arguments to pass to the supplier function
+        :param kwargs: keyword arguments to pass to the supplier function
+        """
+        self._username = "" if username is None else username
```

Comment:
If this is meant to be internal, and private ideally it would be self.__username__.

Same for __passsword__

Reply:
Not sure I understand you here, _username/ __username is the python way of saying a member is private, isn't it?
also self.username is a property so I can’t change the member to self.username too

Reply:
changed to __password